src/adaptors/tcp_adaptor.c

```diff
@@ -782,6 +965,14 @@ static void qdr_tcp_open_server_side_connection(qdr_tcp_connection_t* tc)
 
     // This attach passes the ownership of the delivery from the core-side connection and link
     // to the adaptor-side outgoing connection and link.
+    uint64_t i_conn_id = 0;
+    uint64_t i_link_id = 0;
+    uint64_t i_dlv_id = 0;
+    if (!!tc->initial_delivery) {
+        i_conn_id = tc->initial_delivery->conn_id;
```

Comment:
Minor point, but why are these initialised in a separate conditional block up here, when they seem only to be used in the subsequent block for the identical condition and are not used in the qdr_link_first_attach() call that is the only thing separating these blocks? Makes it easier to read if we just declared these just before use I think.

Reply:
qdr_link_first_attach conditionally "adjusts" the delivery's identity. This code is conditionally memorizing what the initial delivery had before the adjustment.
The log statement giving identical [Cx]{Lx][Dx] values before and after was pretty lame.

Reply:
Sorry, I still don't get it. The i_dlv_id is not referenced in the qdr_link_first_attach call.